pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -422,6 +425,19 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []v1.Volume, namespace s
 				continue
```

Comment:
What about this, would be acceptable to make the pod unschedulable?

We currently fail if the PVC name is empty, but we count the volume if the PVC is invalid.

 I think counting this volume is not correct, since at this point we're not sure whether the volume belongs to the running predicate (i.e., it could be a EBS volume being counted by the GCEPD predicate).

Reply:
This was an error condition we added a while ago to handle the case where users may delete PVCs while Pods were still running and using them.  But now that we have StorageProtection, do we need this check?  cc @jsafrane

Reply:
> This was an error condition we added a while ago to handle the case where users may delete PVCs while Pods were still running and using them

Hmm, scheduler ignores pods that are running. IMO this check covers users (and StatefulSets) creating Pods before PVCs.

Reply:
My argument is that StorageProtection handles 1).  And scheduler prevents 2). If immediate binding, predicate fails. If delayed binding, we count further down.

Reply:
PVC protection can be forced off, so scheduler can encounter a pod that refers to PVC that does not exist. If `filterVolumes` is counting running pods, I would suggest not to count the volume - user has broken something and miscalculations are IMO OK. If `filterVolumes` checks a newly scheduled pod, I'd return error and prevented the pod from scheduling.

Reply:
I like this approach, it addresses my concern.

Reply:
We don't strictly need separate behavior for counting separate pods vs to be scheduled pod. The volume binding predicate checks for bound pvcs

Reply:
Hm, yeah, it does make sense for this predicate to only make the pod unschedulable because of volume limits; and leave the pvc binding checks for the volume binding predicate.

Updated the code, PTAL.